db/compaction/compaction_service_job.cc

```diff
@@ -28,6 +29,13 @@ CompactionJob::ProcessKeyValueCompactionWithCompactionService(
 
   const Compaction* compaction = sub_compact->compaction;
   CompactionServiceInput compaction_input;
+  Status s = GetLatestOptionsFileName(dbname_, db_options_.env,
+                                      &compaction_input.options_file);
+  if (!s.ok()) {
```

Comment:
Can the file be deleted before remote instance reads it?

Reply:
Ah good question. Yes, if there's options change between RemoteCompaction enqueue and this line, the file can be missing and we will have failure here.

https://github.com/facebook/rocksdb/blob/71e38dbe25253dc19e0ca0214ee8244d6731ac76/db/db_impl/db_impl.cc#L5443-L5450

Once the file is read, we don't need the file to exist anymore while the job is running. Wondering how often we could run into this issue in production and something that we can tolerate. To even lower the chance, I think we can keep the last N OPTIONS file, which can be controlled behind an option. (for the users who change options often)

Reply:
Could you simply retry? The OPTIONS file should only change when `SetDBOptions` is called, which should be pretty rare.

Reply:
>Could you simply retry?

Which one? I am not sure if retry on either of them would actually be helpful.

1. For `GetLatestOptionsFileName()` in the primary host, I believe that it only fails if no OPTIONS file is found at all.
2. For `LoadOptionsFromFile(file_name_from_the_payload)` in the remote worker, retry won't succeed if the file is gone. 

Maybe I'm wondering if we should just call `GetLatestOptionsFileName()` from the remote worker and always uses the latest options instead of passing the options file name over the payload.

Reply:
Just updated the PR to always load the latest options file instead of passing the file name in the payload. With a single retry, compaction failure due to missing OPTION file should be extremely rare.